keras_cv/models/mobilenet_v3.py

```diff
@@ -0,0 +1,534 @@
+            network.  If provided, `num_classes` must be provided.
+        weights: one of `None` (random initialization), or a pretrained weight file
+            path.
+        num_classes: optional number of classes to classify images into, only to be
+            specified if `include_top` is True, and if no `weights` argument is
+            specified.
+        pooling: optional pooling mode for feature extraction
+            when `include_top` is `False`.
+            - `None` means that the output of the model will be the 4D tensor output
+                of the last convolutional block.
+            - `avg` means that global average pooling will be applied to the output
+                of the last convolutional block, and thus the output of the model will
+                be a 2D tensor.
+            - `max` means that global max pooling will be applied.
+        dropout_rate: a float between 0 and 1 denoting the fraction of input units to
+            drop, defaults to 0.2.
+        classifier_activation: the activation function to use, defaults to softmax.
+        include_rescaling: whether or not to Rescale the inputs.If set to True,
+            inputs will be passed through a `Rescaling(scale=1.0 / 127.5, offset=-1.0)`
+            layer, defaults to True.
+        minimalistic: in addition to large and small models this module also
+            contains so-called minimalistic models, these models have the same
+            per-layer dimensions characteristic as MobilenetV3 however, they don't
+            utilize any of the advanced blocks (squeeze-and-excite units, hard-swish,
+            and 5x5 convolutions). While these models are less efficient on CPU, they
+            are much more performant on GPU/DSP.
+        name: (Optional) name to pass to the model. Defaults to "MobileNetV3".
+
+    Returns:
+        A `keras.Model` instance.
+
+    Raises:
+        ValueError: if `weights` represents an invalid path to weights file and is not
+            None.
+        ValueError: if `include_top` is True and `num_classes` is not specified.
+    """
+    if weights and not tf.io.gfile.exists(weights):
+        raise ValueError(
+            "The `weights` argument should be either "
+            "`None` or the path to the weights file to be loaded. "
+            f"Weights file not found at location: {weights}"
+        )
+
+    if include_top and not num_classes:
+        raise ValueError(
+            "If `include_top` is True, "
+            "you should specify `num_classes`. "
+            f"Received: num_classes={num_classes}"
+        )
+
+    if minimalistic:
+        kernel = 3
+        activation = layers.ReLU()
+        se_ratio = None
+    else:
+        kernel = 5
+        activation = HardSwish()
+        se_ratio = 0.25
+
+    inputs = layers.Input(shape=input_shape)
+
+    x = inputs
+
+    if include_rescaling:
+        x = layers.Rescaling(scale=1.0 / 127.5, offset=-1.0)(x)
```

Comment:
any reason not to just rescale by 1/255?

Reply:
The original `keras.applications` uses this rescaling in the preprocessing step, please see:

https://github.com/keras-team/keras/blob/v2.9.0/keras/applications/mobilenet_v2.py#L517

https://github.com/keras-team/keras/blob/07e13740fd181fc3ddec7d9a594d8a08666645f6/keras/applications/imagenet_utils.py#L191-L194

Thus, I thought to go ahead with this

Reply:
Got you